lib/phoenix/router/console_formatter.ex

```diff
@@ -19,7 +22,28 @@ defmodule Phoenix.Router.ConsoleFormatter do
     ])
   end
 
-  defp format_endpoint(nil, _), do: ""
+  defp expand_dynamic_routes(routes) do
+    Enum.flat_map(routes, fn route ->
+      Code.ensure_loaded(route.plug)
+
+      if function_exported?(route.plug, :phoenix_routes, 1) do
+        route.plug_opts
+        |> route.plug.phoenix_routes()
```

Comment:
This is not a contract in Phoenix anywhere else, so this should be removed in favor of formatted_routes. I'd argue it shuold most likely be moved to Phoenix.Router implementation.

Reply:
Sorry, that was a mistake that I left it in. It should be replaced with `formatted_routes`. Will do that now.

I thought about pulling it into `formatted_routes` in `Phoenix.Router`, but it also seemed like a potential benefit for it to be expanded here. In the case that an implementer of this returned a route with a plug that could be expanded, they wouldn't have to know to do that. I can see a case for both, so I'll move it into `Phoenix.Router` 👍